plugins/module_utils/foreman_helper.py

```diff
@@ -640,7 +640,7 @@ def _delete_entity(self, resource, current_entity, params):
         entity = self.resource_action(resource, 'destroy', payload)
 
         # this is a workaround for https://projects.theforeman.org/issues/26937
-        if entity and 'error' in entity and 'message' in entity['error']:
+        if entity and isinstance(entity, dict) and 'error' in entity and 'message' in entity['error']:
             self.fail_json(msg=entity['error']['message'])
```

Comment:
The variable name `entity` feels kind of wrong here.

I know, that's not primarily the scope of this PR.

Reply:
Yeah, scope creep. The patch here is just for https://projects.theforeman.org/issues/29403

Reply:
:see_no_evil: